Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1077,6 +1077,24 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: EnumCaseElementSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let unexpectedBetweenNameAndParameterClause = node.unexpectedBetweenNameAndParameterClause,
+      let genericParameter = unexpectedBetweenNameAndParameterClause.compactMap({ $0.as(GenericParameterClauseSyntax.self) }).only
+    {
+      addDiagnostic(
+        genericParameter,
+        .genericParamCantBeUsedInEnumCaseDecl,
+        handledNodes: [unexpectedBetweenNameAndParameterClause.id]
+      )
```

Comment:
The C++ version have a Fix-Its. 
Not sure if we can do the same as we need to modify the enum decl.

Reply:
Hmm, it’s not easily possible in the current design where we assume all nodes for the Fix-It to already be present in the syntax tree so that we just need to flip the token presence. 

I would say that it’s good enough for now but adding support for these kinds of Fix-Its might be useful in general. What we need for that is, IMO, a new `Fixit.Change` that takes a syntax node, a key path within that node and a replacement node, i.e. something like `case replace<NodeType, ChildType>(node: NodeType, path: KeyPath<NodeType, ChildType>, child: ChildType`) – and yes I see the irony of declaring a generic enum case here, you would need to come up with some design that actually compiles, I haven’t thought about it too much yet.

I think this kind of Fix-It change might also be useful for macro authors. IIRC it came up once or twice but was never super important.

Reply:
Let's do that in another PR!

Opened an issue: https://github.com/apple/swift-syntax/issues/2298